app/controllers/prior_authority/additional_costs_controller.rb

```diff
@@ -0,0 +1,46 @@
+module PriorAuthority
+  class AdditionalCostsController < PriorAuthority::BaseController
+    def edit
+      crime_application = PriorAuthorityApplication.find(params[:application_id])
+      all_costs = BaseViewModel.build(:additional_cost, crime_application, 'additional_costs')
+      index = all_costs.index do |model|
+        model.id == params[:id]
+      end
+
+      item = all_costs[index]
+      form = AdditionalCostForm.new(crime_application:, item:, **item.form_attributes)
+
+      render locals: { crime_application:, item:, form:, index: }
+    end
+
+    def update
+      crime_application = PriorAuthorityApplication.find(params[:application_id])
+      all_costs = BaseViewModel.build(:additional_cost, crime_application, 'additional_costs')
+      index = all_costs.index do |model|
+        model.id == params[:id]
+      end
+
+      item = all_costs[index]
```

Comment:
would be interested in understanding why you use `index` over `detect`

Reply:
It's only because we need to have `index` as a variable in the views so that, as per the current designs, we can use it in the page title (e.g. "Additional Cost 1"). Since we needed to get the index anyway, I thought we might as well get that, then use it to retrieve the item, rather than go through the list of items twice, once for the item and once for the index.

(This is probably throwaway code - I'm sure when we come to build the form in earnest there will be more scrutiny of how additional cost items are referred to, and the `index + 1` approach may go out the window)